### PR TITLE
[Merged by Bors] - chore(Data/Finset/Basic): Depend on less order theory

### PR DESCRIPTION
**Pull request by YaelDillies**
Move `Finset.biUnion` and `Finset.disjiUnion` to a new file so that `Data.Finset.Basic` doesn't depend on that much order theory.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Set/Finite.lean [hunk 1]
```diff
@@ -484,7 +485,7 @@ instance fintypeMap {α β} [DecidableEq β] :
 #align set.fintype_map Set.fintypeMap
 
 instance fintypeLTNat (n : ℕ) : Fintype { i | i < n } :=
-  Fintype.ofFinset (Finset.range n) <| by simp
+  Fintype.ofFinset ⟨_, Multiset.nodup_range n⟩ <| by simp
```
Comment (fpvandoorn) [inline Mathlib/Data/Set/Finite.lean:488]: Why was this changed?

File: Mathlib/Data/Set/Finite.lean [hunk 2]
```diff
@@ -1356,8 +1357,8 @@ noncomputable def Infinite.natEmbedding (s : Set α) (h : s.Infinite) : ℕ ↪
 #align set.infinite.nat_embedding Set.Infinite.natEmbedding
 
 theorem Infinite.exists_subset_card_eq {s : Set α} (hs : s.Infinite) (n : ℕ) :
-    ∃ t : Finset α, ↑t ⊆ s ∧ t.card = n :=
-  ⟨((Finset.range n).map (hs.natEmbedding _)).map (Embedding.subtype _), by simp⟩
+    ∃ t : Finset α, ↑t ⊆ s ∧ t.card = n := by
+  obtain ⟨t, ht, rfl⟩ := hs.to_subtype.exists_subset_card_eq n; exact ⟨t.map $ .subtype _, by simp⟩
```
Comment (fpvandoorn) [inline Mathlib/Data/Set/Finite.lean:1361]: ^

**Review by fpvandoorn — COMMENTED** [review]
Please move the text about `Finset.biUnion` in the module doc as well.

Please add at least one sentence of doc in `Finset.Piecewise`.

File: Mathlib/Data/Set/Finite.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Data/Set/Finite.lean:488]: Whoops, copy-pasta

**Comment by fpvandoorn** [comment]
bors merge
